server/sonar-server/src/test/java/org/sonar/server/settings/ws/ListDefinitionsActionTest.java

```diff
@@ -135,6 +135,25 @@ public void return_settings_definitions_with_minimum_fields() {
     assertThat(definition.getMultiValues()).isFalse();
     assertThat(definition.getOptionsCount()).isZero();
     assertThat(definition.getFieldsCount()).isZero();
+    assertThat(definition.hasDeprecatedKey()).isFalse();
+  }
+
+  @Test
+  public void return_settings_definitions_with_deprecated_key() {
+    setUserAsSystemAdmin();
+    propertyDefinitions.addComponent(PropertyDefinition
+      .builder("foo")
+      .name("Foo")
+      .deprecatedKey("deprecated")
+      .build());
+
+    ListDefinitionsWsResponse result = newRequest();
```

Comment:
suggestion: separate the 3 phases of a test with AAA: Arrange, Act, Assert. Your line `ListDefinitionsWsResponse result = newRequest();` would then be separated from other lines. I find it's an easy to think about tests. It's different for `ExpectedException` though